In [7]:
import pandas as pd
df = pd.read_csv("final_data/wineries.csv")
df.head()

country           province             region_1           region_2  \
0     Italy  Sicily & Sardinia                 Etna                NaN   
1  Portugal              Douro                  NaN                NaN   
2        US             Oregon    Willamette Valley  Willamette Valley   
3        US           Michigan  Lake Michigan Shore                NaN   
4        US             Oregon    Willamette Valley  Willamette Valley   

                winery  winery_id  
0              Nicosia          0  
1  Quinta dos Avidagos          1  
2            Rainstorm          2  
3           St. Julian          3  
4         Sweet Cheeks          4

### Fixing "country" and "province" Values

#### Add in missing "country" and "province" values
First, identify the wineries with missing "country" and "province" information and verify that they're the same

In [19]:
idx_missing_location = df[df['country'].isnull()].index
missing_countries = df["winery"].loc[idx_missing_location]

idx_missing_location = df[df['province'].isnull()].index
missing_provinces = df["winery"].loc[idx_missing_location]

list(missing_countries)==list(missing_provinces)

True

Then, look up the winery names and find the true values.

In [20]:
df["winery"].loc[idx_missing_location]

838                   Gotsa Family Wines
2636                   Barton & Guestier
3460     Kakhetia Traditional Winemaking
6715                            Tsililis
6837                            Ross-idi
7554                             Orbelus
8931                           St. Donat
9714                      Familia Deicas
10008                     Bartho Eksteen
13507                       Stone Castle
13508                     Teliani Valley
13511                          Undurraga
13912                      Mt. Beautiful
16794                         Neumeister
16956                          Bachelder
18523                            Chilcas
18714                          Santa Ema
19347                             Psagot
20966                     Newton Johnson
21470                      Ktima Voyatzi
25333                            Lismore
25334                            Logodaj
25958                      Somlói Vándor
27408                  Les Frères Dutruy
28854           

In [23]:
# Find list of provinces based on country 
country =  "Georgia"
idx_country = df[df['country']==country].index
print(set(df["province"].loc[idx_country]))

{'Georgia', 'Kakheti'}

| Winery    | Index | Country | Province |
| -------- | ------- | ------- | ------- |
| Gotsa Family Wines | 838 | Georgia | Kvemo Kartli |
| Barton & Guestier | 2636 | France | Gironde |
| Kakhetia Traditional Winemaking | 3460 | Georgia | Kakheti |
| Tsililis |6715 | Greece | Thessaly |
| Ross-idi | 6837 | Bulgaria | Nikolaevo |
| Orbelus | 7554 | Bulgaria | Blagoevgrad |
| St. Donat | 8931 | Hungary | Veszprem |
| Familia Deicas | 9714 | Uruguay | Canelones |
| Bartho Eksteen | 10008 | 
| Stone Castle | 13507 | Kosovo
| Teliani Valley | 13508 | 
| Undurraga | 13511 | 
| Mt. Beautiful | 13912 | 
| Neumeister | 16794 | 
| Bachelder | 16956 | 
| Chilcas | 18523 | 
| Santa Ema | 18714 | 
| Psagot | 19347 | 
| Newton Johnson | 20966 | 
| Ktima Voyatzi | 21470 | 
| Lismore | 25333 | South Africa
| Logodaj | 25334 | 
| Somlói Vándor | 25958 | Hungary
| Les Frères Dutruy | 27408 | Switzerland
| El Capricho | 28854 | Uruguay
| Amiran Vepkhvadze | 29464 | Georgia
| Büyülübağ | 29464 | Turkey


#### Check for "country" and "province" duplicates

### Cleaning "region_1" and "region_2" values

#### Check for duplicates